heidrun/TN_mods.rb

```diff
@@ -0,0 +1,149 @@
+    alternative record.field('mods:titleInfo')
+                      .match_attribute(:type, 'alternative')
+                      .field('mods:title')
+
+    collection :class => DPLA::MAP::Collection,
+               :each => record.field('mods:relatedItem')
+               				  .match_attribute(:type, 'host')
+               				  .match_attribute(:displayLabel, 'Project')
+               				  .field('mods:titleInfo', 'mods:title'),
+               :as => :coll do
+      title coll
+    end
+
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.field('mods:name')
+                        .select { |name| name['mods:role'].map(&:value).include?('contributor') },
+                :as => :contrib do
+      providedLabel contrib.field('mods:namePart')
+    end
+    
+    creator :class => DPLA::MAP::Agent,
+            :each => record.field('mods:name')
+                    .select { |name| name['mods:role'].map(&:value).include?('creator') },
+            :as => :creator_role do
+      providedLabel creator_role.field('mods:namePart')
+    end
+
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.field('mods:originInfo'),
+         :as => :created do
+      providedLabel created.field('mods:dateCreated')
+    end
+
+    description record.field('mods:abstract')
+
+    extent record.field('mods:physicalDescription', 'mods:extent')
+
+    # non-DCMIType values from type will be handled in enrichment
+    dcformat record.field('mods:physicalDescription', 'mods:form')
+
+    genre record.field('mods:genre').match_attribute(:authority 'aat')
+
+    identifier record.field('mods:identifier')
+
+    language :class => DPLA::MAP::Controlled::Language,
+             :each => record.field('mods:language', 'mods:languageTerm')
+             				.match_attribute(:type 'code')
+             				.match_attribute(:authority 'iso639-2b'),
+             :as => :lang do
+      prefLabel lang
+    end
+
+    spatial :class => DPLA::MAP::Place,
+            :each => record.field('mods:subject', 'mods:geographic'),
+            :as => :place do
+      providedLabel place
+    end
+    
+    #They have coordinates as well. I wasn't sure if I should add a second spatial mapping or include this in the previous. I decided to add a second. Is that correct?
+    spatial :clas => DPLA::MAP::Place,
+      		:each => record.field('mods:subject', 'mods:cartographics', 'mods:coordinates'),
+      		:as => :lat do
+      wgs84_pos:lat lat
+    end
+
+    publisher :class => DPLA::MAP::Agent,
+              :each => record.field('mods:originInfo'),
+              :as => :publisher do
+      providedLabel publisher.field('mods:publisher')
+    end
+    
+    #totally a guess as to how to not map properties with certain attributes. for relation
+    relation record.field('mods:relatedItem')
+    				.reject_attribute(:type 'isReferencedBy')
+    				.reject_attribute(:type 'references')
+    				.field('mods:titleInfo', 'mods:title')
```

Comment:
Do `relation`, `isReplacedby`, and `replaces` need to be *either* the `mods:title` or `mods:url`, or *both*?